packages/design-mui/src/components/MuiFilters/MuiFilters.tsx

```diff
@@ -29,6 +31,9 @@ export const MuiFilters: React.FC<XCollectionProps> = ({
         }
         return filter.value !== "" && (filter.value as string[]).length !== 0
       })
+      if (queryFilters.length) {
```

Comment:
This seems to me like it should fire when `queryFilters` changes it's values (perhaps based on some diff), not when ever their are queryFilters.